src/modules/service-catalog/ServiceCatalogItemPage.tsx

```diff
@@ -97,43 +34,69 @@ export function ServiceCatalogItemPage({
   brandId,
 }: ServiceCatalogItemPageProps) {
   const serviceCatalogItem = useServiceCatalogItem(serviceCatalogItemId);
-  const { requestFields, handleChange } = useItemFormFields(
-    serviceCatalogItem,
-    baseLocale
-  );
+  const {
+    requestFields,
+    associatedLookupField,
+    setRequestFields,
+    handleChange,
+  } = useItemFormFields(serviceCatalogItem, baseLocale);
   const { t } = useTranslation();
+  const handleRequestSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const response =
+      serviceCatalogItem &&
```

Comment:
Isn't there a case where we will call this function when `serviceCatalogItem` is null, right? We are showing the button only if it is not null.

If this is the case I would just add a `if (!serviceCatalogItem) { return; }` at the beginning of the function because using this pattern when the response can be null makes the code difficult to read with a lot of ifs.

I also think we need to handle these edge cases better, if we cannot fetch the custom object we are just not showing the form, and if we don't find the lookup field we don't do anything. These scenarios are blocking because we cannot do anything without this data. We can eventually address this in a separate PR where we introduce error handling

Reply:
Ok changed to your suggestion. I think I already created some task for adding ErrorBoundary if not I will create and I will add it everywhere it is needed.